### PR TITLE
add ut of client.go

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What this PR does / why we need it**:
add ut of client.go
**Which issue(s) this PR fixes**:
a part of #104

### CODE REVIEW

File: pkg/controller/client_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+	"gotest.tools/assert"
+
+	"kmesh.net/kmesh/pkg/bpf"
+	"kmesh.net/kmesh/pkg/controller/envoy"
+	"kmesh.net/kmesh/pkg/nets"
+)
+
+type MockDiscovery struct {
```
Comment (LiZhenCheng9527) [inline pkg/controller/client_test.go:38]: This function will be processed later

**Comment by LiZhenCheng9527** [comment]
cancel /hold

File: pkg/controller/client_test.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+			func(_ *XdsClient) error {
+				// more than 2 link failures will result in a long test time
+				if iteration < 2 {
+					iteration++
+					return errors.New("cant connect to client")
+				} else {
+					return nil
+				}
+			})
+		err := utClient.recoverConnection()
```
Comment (hzxuzhonghu) [inline pkg/controller/client_test.go:118]: please do some more meaningful test.

**Review by hzxuzhonghu — COMMENTED** [review]
I am super sure these tests can be flaky.

File: pkg/controller/client_test.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+		streamPatches := gomonkey.NewPatches()
+		defer streamPatches.Reset()
+		streamPatches.ApplyMethod(reflect.TypeOf(utClient.AdsStream), "AdsStreamProcess",
+			func(_ *envoy.AdsStream) error {
+				if iteration < 2 {
+					return errors.New("stream recv failed")
+				} else {
+					utClient.cancel()
+					return nil
+				}
```
Comment (hzxuzhonghu) [inline pkg/controller/client_test.go:168]: add some comments about this, it is confusing

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**